File: src/main/java/org/smudge/petstore/PetService.java
package org.smudge.petstore;

import org.springframework.stereotype.Service;

import java.util.Collections;
import java.util.HashMap;
import java.util.Map;

@Service
public class PetService {
    //Pet id to Pet
    private Map<Integer, Pet> pets = new HashMap<>();

    public PetService(){
        this.pets.put(0, new Pet(0, 0, "Smudge", "cat"));
        this.pets.put(1, new Pet(1, 1, "Tony", "cat"));
        this.pets.put(2, new Pet(1, 2, "Pepper", "bird"));
        this.pets.put(3, new Pet(2, 3, "Jack", "dog"));
    }

    public Pet getPet(int id) {
        return pets.get(id);
    }

    public Pet createPet(int ownerId, String name, String type) {
        Pet pet = new Pet(ownerId, getNextKey(), name, type);
        this.pets.put(pet.getId(),pet);
        return pet;
    }

    private int getNextKey() {
        return Collections.max(this.pets.keySet())+1;
    }

    public Pet changeName(int petId, String newName) {
        Pet pet = getPet(petId);
        pet.setName(newName);
        return pet;
    }
}

File: src/main/java/org/smudge/petstore/PetService.kt
package org.smudge.petstore


import org.springframework.beans.factory.annotation.Autowired
import org.springframework.stereotype.Service

@Service
class PetService (
        @Autowired
        private val repo: PetRepository
        ) {

    fun getPet(id: Int): Pet {
        return repo.findById(id).orElse(null)?.toBusinessObject()
                ?: throw Exception("No pet found with id $id")
    }

    fun createPet(id: Int,ownerId: Int, name: String, type: String): Pet {
        val newPet = PetEntity(0,ownerId,name,type)
        repo.save(newPet)
        return newPet.toBusinessObject()
    }

    fun changeName(petId: Int, ownerId: Int, newName: String, type: String): Pet {
        val pet = repo.findById(petId).get()
        pet.name = newName
        repo.save(pet)
        return pet.toBusinessObject()
    }


}